examples/fedllm-finetune/README.md

```diff
@@ -0,0 +1,140 @@
+However, despite the fact that more data typically leads to improved performance, there is a concerning prospect that the supply of high-quality public data will deplete within a few years.
+Federated LLM training could unlock access to an endless pool of distributed private data by allowing multiple data owners to collaboratively train a shared model without the need to exchange raw data.
+
+This introductory example conducts federated instruction tuning with pretrained [LLama2](https://huggingface.co/openlm-research) models on [Alpaca-GPT4](https://huggingface.co/datasets/vicgalle/alpaca-gpt4) dataset.
+We use [Flower Datasets](https://flower.dev/docs/datasets/) to download, partition and preprocess the dataset.
+The fine-tuning is done using the [🤗PEFT](https://huggingface.co/docs/peft/en/index) library.
+We use Flower's Simulation Engine to simulate the LLM fine-tuning process in federated way,
+which allows users to perform the training on a single GPU.
+
+## Environments Setup
+
+Start by cloning the code example. We prepared a single-line command that you can copy into your shell which will checkout the example for you:
+
+```shell
+git clone --depth=1 https://github.com/adap/flower.git && mv flower/examples/fedllm-finetune . && rm -rf flower && cd fedllm-finetune
+```
+
+This will create a new directory called `fedllm-finetune` containing the following files:
+
+```
+-- README.md           <- Your're reading this right now
+-- main.py             <- Start fed-LLM simulation
+-- client.py           <- Flower client constructor
+-- model.py            <- Model build
+-- dataset.py          <- Dataset and tokenizer build
+-- utils.py            <- Utility functions
+-- test.py             <- Test pre-trained model
+-- conf/config.yaml    <- Configuration file
+-- requirements.txt    <- Example dependencies
+```
+
+### Installing dependencies
+
+Project dependencies are defined in `requirements.txt`. Install them with:
+
+```shell
+pip install -r requirements.txt
+```
+
+## Run LLM Fine-tuning
+
+### Run with `start_simulation()`
+
+With an activated Python environment, run the example with default config values. The config is in `conf/config.yaml` and is loaded automatically.
+
+```bash
+# Run with default config
+python main.py
+```
+
+This command will run FL simulations with an 8-bit [OpenLLaMA 3Bv2](https://huggingface.co/openlm-research/open_llama_3b_v2) model involving 2 clients per rounds for 100 FL rounds. You can override configuration parameters directly from the command line. Below are a few settings you might want to test:
+
+```bash
+# Use OpenLLaMA-7B instead of 3B and 4-bits quantization
+python main.py model.name="openlm-research/open_llama_7b_v2" model.quantization=4
+
+# Run for 50 rounds but increasing the fraction of clients that participate per round to 25%
+python main.py num_rounds=50 fraction_fit.fraction_fit=0.25
```

Comment:
Wouldn't this result in just 1 client per round?

Reply:
I think part of the confusion is that we have one strategy for using simulation (which is how this example was first designed, and therefore most of the `REAMDE.md`) and another one for the `Flower Next` demo using `app.py`.